src/Storages/MergeTree/KeyCondition.h

```diff
@@ -485,6 +485,6 @@ class KeyCondition
     bool strict;
 };
 
-String extractFixedPrefixFromLikePattern(const String & like_pattern);
+String extractFixedPrefixFromLikePattern(const String & like_pattern, bool perfect_prefix_match = false);
```

Comment:
Please, let's remove the default parameter to minimize surprises and rewrite callers like this:

```cpp
extractFixedPrefixFromLikePattern(pattern, /*perfect_prefix_match*/ false);
extractFixedPrefixFromLikePattern(pattern, /*perfect_prefix_match*/ true);
```